### ISSUE TITLE
[3.7.2] Slow admin after login

### ISSUE THREAD

**Issue by tranduyhung** [joomla/joomla-cms#16349]
### Steps to reproduce the issue

Log into admin, access somewhere else (user list, article list,...) or just refresh the dashboard. Please see additional comments for more info.

### Expected result

The next page is loaded with normal speed.

### Actual result

The next page takes about 20 - 30 seconds to load.

### System information (as much as possible)

PHP Built On: Linux cloud.rexmedia.nl 3.10.0-514.16.1.el7.x86_64 #1 SMP Wed Apr 12 15:04:24 UTC 2017 x86_64
Database Version: 5.5.52-MariaDB
Database Collation: utf8_general_ci
Database Connection Collation: utf8mb4_general_ci
PHP Version: 7.0.19
Web Server: Apache
WebServer to PHP Interface: cgi-fcgi
Joomla! Version: Joomla! 3.7.2 Stable [ Amani ] 22-May-2017 09:46 GMT
Joomla! Platform Version: Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT

### Additional comments

This problem happens on a site of my friend. I can't reproduce this problem on my site, maybe because I have less extensions installed (see the list below).

When turning debug on, I've found out that the problem is in preloadComponents of JAccess class.

![screenshot from 2017-05-30 08-46-07](https://cloud.githubusercontent.com/assets/503262/26566464/a04564f6-451d-11e7-89d8-340cc17a04d1.jpg)

The problem only happens after logging in and access another page, after that it is gone.

![screenshot from 2017-05-30 08-47-01](https://cloud.githubusercontent.com/assets/503262/26566501/e27f1902-451d-11e7-8262-97322ab5698b.jpg)

This is not an issue with database query, the query for assets table only takes about 0.15 ms:

```sql
SELECT `id`,`name`,`rules`,`parent_id`

  FROM `clmp_assets`

  WHERE `name` IN ('com_admin','com_advancedmodules','com_ajax','com_akeeba','com_associations','com_banners','com_cache','com_categories','com_checkin','com_config','com_contact','com_contactenhanced','com_content','com_contenthistory','com_cpanel','com_customfilters','com_easyquickicons','com_fields','com_finder','com_gantry','com_installer','com_jce','com_jdidealgateway','com_joomlaupdate','com_k2','com_languages','com_login','com_mailto','com_media','com_menus','com_messages','com_modules','com_newsfeeds','com_plugins','com_postinstall','com_quix','com_redirect','com_regularlabsmanager','com_rsform','com_search','com_sigpro','com_sisowpaymentinovations','com_tags','com_tcpdf','com_templates','com_uniterevolution','com_users','com_virtuemart','com_virtuemart_allinone','com_vmexcellikeinput','com_weblinks','com_wrapper') OR `parent_id` = 0
```

All other queries are pretty fast, nothing is unusual.

I've tried to debug in JAccess but still not found where the problem is. I am not sure if this problem happens just because we have many extensions on the site.

The site has 3 access levels and 8 user groups.

**Comment by tonypartridge** [comment]
How many components do you have? I'm wondering as a test can you disable all 3rd party components and try again? Enabling 1 by 1 to see if it's caused by a single component?

**Comment by brianteeman** [comment]
@tonypartridge they are listed in the Select query above

**Comment by tonypartridge** [comment]
@brianteeman I didn't see it scrolled (using my phone). Still my suggestion is Still valid I believe.

**Comment by brianteeman** [comment]
@tonypartridge yes it would be a sensible test. Looking at the list I have my strong suspicions and I would start at the end of the list

**Comment by tonypartridge** [comment]
Agreed!

**Comment by andrepereiradasilva** [comment]
hi, your issue doesn't seem to be related with JAccess::preloadComponents

Read again the debug profile line, it says "**Before** JAccess::preloadComponents".

The delay you have is happening after "afterLoad" event (in https://github.com/joomla/joomla-cms/blob/staging/index.php#L43), and **before** JAccess::preloadComponents" is executed (in https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/access/access.php#L403).

**Comment by tonypartridge** [comment]
How did I miss that @andrepereiradasilva  clearly I was very tired...

**Comment by brianteeman** [comment]
Welcome back @andrepereiradasilva

**Comment by tranduyhung** [comment]
My friend sent me a backup to let me try on my localhost, after I disabled the component one by one, I found out that Contact Enhanced component is the reason of this problem. If I disable it, the problem disappears. I've asked my friend to talk to Contact Enhanced's developer.

Before closing this issue, could some one give me more info how a component could slow the site down when it is not executed? I have checked and it doesn't have a button plugin enabled which doesn't involve.
